File: app/src/main/java/com/datarockets/mnchkn/activities/players/di/PlayersListComponent.java
package com.datarockets.mnchkn.activities.players.di;

import com.datarockets.mnchkn.activities.players.PlayersListActivity;
import com.datarockets.mnchkn.activities.players.PlayersListPresenter;
import com.datarockets.mnchkn.di.ActivityScope;
import com.datarockets.mnchkn.di.AppComponent;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = PlayersListModule.class)
public interface PlayersListComponent {
    void inject(PlayersListActivity playersListActivity);
    PlayersListPresenter getPlayersListPresenter();
}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/players/di/PlayersListComponent.kt
package com.datarockets.mnchkn.activities.players.di

import com.datarockets.mnchkn.activities.players.PlayersListActivity
import com.datarockets.mnchkn.activities.players.PlayersListPresenter
import com.datarockets.mnchkn.di.ActivityScope
import com.datarockets.mnchkn.di.AppComponent

import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(PlayersListModule::class))
interface PlayersListComponent {
    fun inject(playersListActivity: PlayersListActivity)
    val playersListPresenter: PlayersListPresenter
}